app/models/project.rb

```diff
@@ -91,6 +95,23 @@ def is_admin?(user)
     user && (admin_group.users.member?(user) || user.admin)
   end
 
+  def can_join?(user)
+    open_membership && !is_member?(user)
+  end
+
+  def can_leave?(user)
+    is_member?(user) && user.id != user_id
+  end
+
+  def check_box_disabled?(obs, user)
+    !accepting_observations || (obs.user != user &&
+                                !is_member?(user))
+  end
+
+  def enabled?
```

Comment:
This is just a personal preference: `project.enabled?` by itself doesn't tell me sufficiently what it means. I would honestly prefer in-lining this over the convenience method.  `project.open_membership && project.accepting_observations` is completely clear.  I'm sure it would be possible to elaborate a more descriptive method name than "enabled?" but it might end up being something like "open_and_accepting_observations?" which is scarcely shorter than just in-lining it! :)

Reply:
Reasonable.  I used to have a comment about this getting more complicated due to potentially adding date, location, or taxon constraints.  I'm not convinced this is really the direction anymore, so I agree with getting rid of this.

Reply:
Done